### PR TITLE
[Merged by Bors] - feat: two missing lemmas about restricting continuous maps

### PR DESCRIPTION
**Pull request by ADedecker**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dagurtomas** [comment]
Looks good to me -- works well in the implementation of `LocallyConstant.piecewise'` in #6589.

File: Mathlib/Topology/Constructions.lean [hunk 1]
```diff
@@ -1100,8 +1100,8 @@ theorem closure_subtype {x : { a // p a }} {s : Set { a // p a }} :
 #align closure_subtype closure_subtype
 
 theorem continuousAt_codRestrict_iff {f : α → β} {t : Set β} (h1 : ∀ x, f x ∈ t) {x : α} :
```
Comment (ocfnash) [inline Mathlib/Topology/Constructions.lean:1102]: I recognise you're just golfing the proof but while you're here how about adding `simp`
```suggestion
@[simp]
theorem continuousAt_codRestrict_iff {f : α → β} {t : Set β} (h1 : ∀ x, f x ∈ t) {x : α} :
```

File: Mathlib/Topology/Constructions.lean [hunk 2]
```diff
@@ -1123,6 +1123,14 @@ theorem Continuous.codRestrict {f : α → β} {s : Set β} (hf : Continuous f)
   hf.subtype_mk hs
 #align continuous.cod_restrict Continuous.codRestrict
 
+theorem Continuous.restrict {f : α → β} {s : Set α} {t : Set β} (h1 : MapsTo f s t)
```
Comment (ocfnash) [inline Mathlib/Topology/Constructions.lean:1126]:
```suggestion
@[continuity]
theorem Continuous.restrict {f : α → β} {s : Set α} {t : Set β} (h1 : MapsTo f s t)
```

File: Mathlib/Topology/Constructions.lean [hunk 3]
```diff
@@ -1123,6 +1123,14 @@ theorem Continuous.codRestrict {f : α → β} {s : Set β} (hf : Continuous f)
...
+    (h2 : Continuous f) : Continuous (h1.restrict f s t) :=
+  (h2.comp continuous_subtype_val).codRestrict _
+
+theorem Continuous.restrictPreimage {f : α → β} {s : Set β} (h : Continuous f) :
```
Comment (ocfnash) [inline Mathlib/Topology/Constructions.lean:1130]:
```suggestion
@[continuity]
theorem Continuous.restrictPreimage {f : α → β} {s : Set β} (h : Continuous f) :
```

**Review by ocfnash — APPROVED** [review]
Thanks!

bors d+

**Comment by ADedecker** [comment]
bors merge

**Pull request closed without merging**